blog/2020-11-05-HowMuchFasterIsJava15.adoc

```diff
@@ -0,0 +1,105 @@
+conflicts between every entity and every other entity.
+
+== Executive summary
+
+With Java 15, the average improvement is *10.5%* for G1 and *6.2%* for Parallel GC. The difference between the two
+garbage collectors running on JDK 15 is *10.7%* leaning in favor of Parallel GC.
+For more information about difference between various GC algorithms, please see
+https://dzone.com/articles/choosing-the-best-garbage-collection-algorithm-for[this article].
+
+The Parallel GC is the superior GC for OptaPlanner than G1, since the throughput is still the most relevant factor when it comes to garbage collection.
+
+== Results
+
+=== Java 11 vs. Java 15
+
+image::JDK11andJDK15UsingG1GC.svg[]
+
+[#table1]
+.Score calculation count per second with G1 GC
+|===
+| 2+^.^|Cloud balancing 2+^.^|Machine reassignment 2+^.^|Course scheduling 2+^.^|Exam scheduling 2+^.^|Nurse rostering^.^|Traveling Tournament
+|JDK ^|200c ^|800c ^|B1 ^|B10 ^|c7 ^|c8 ^|s2 ^|s3 ^|m1 ^|mh1 ^|nl14
+|OpenJDK 11 >|77,073 >|75,343 >|239,262 >|37,991 >|6,861 >|8,577 >|12,748 >|8,375 >|3,047 >|2,872 >|1,905
+|JDK 15 >|79,275 >|76,276 >|298,923 >|39,883 >|7,694 >|10,049 >|14,438 >|9,421 >|3,384 >|3,155 >|2,015
+|Difference >|2.9% >|1.2% >|25.0% >|5.0% >|12.1% >|17.2% >|13.3% >|12.5% >|11.0% >|9.9% >|5.8%
+|Average 11+^.^|*10.5%*
+|===
+
+image::JDK11andJDK15UsingParallelGC.svg[]
+
+
+.Score calculation count per second with Parallel GC
+|===
+| 2+^.^|Cloud balancing 2+^.^|Machine reassignment 2+^.^|Course scheduling 2+^.^|Exam scheduling 2+^.^|Nurse rostering^.^|Traveling Tournament
+|JDK ^|200c ^|800c ^|B1 ^|B10 ^|c7 ^|c8 ^|s2 ^|s3 ^|m1 ^|mh1 ^|nl14
+|OpenJDK 11 >|95,716 >|91,519 >|313,653 >|29,254 >|8,079 >|10,510 >|14,842 >|9,474 >|3,810 >|3,518 >|2,641
+|JDK 15	>|100,246 >|95,220 >|309,048 >|26,090 >|9,109 >|11,690 >|16,529 >|10,827 >|4,209 >|3,921 >|2,657
+|Difference >|4.7% >|4.0% >|-1.5% >|-10.9% >|12.8% >|11.2% >|11.4% >|14.3% >|10.5% >|11.5% >|0.6%
+|Average 11+^.^|*6.2%*
+|===
+
+
+=== Parallel GC vs. G1 GC on Java 15
+
+image::G1GCandParallelGCUsingJDK15.svg.svg[]
+
+
+[#table3]
+.Comparison of score calculation count per second on Java 15 with Parallel GC and G1 GC being used
+|===
+| 2+^.^|Cloud balancing 2+^.^|Machine reassignment 2+^.^|Course scheduling 2+^.^|Exam scheduling 2+^.^|Nurse rostering^.^|Traveling Tournament
+|Java 15 ^|200c ^|800c ^|B1 ^|B10 ^|c7 ^|c8 ^|s2 ^|s3 ^|m1 ^|mh1 ^|nl14
+|JDK 15 G1GC >|79,275 >|76,276 >|298,923 >|39,883 >|7,694 >|10,049 >|14,438 >|9,421 >|3,384 >|3,155 >|2,015
+|JDK 15	ParallelGC >|100,246 >|95,220 >|309,048 >|26,090 >|9,109 >|11,690 >|16,529 >|10,827 >|4,209 >|3,921 >|2,657
+|Difference >|20.92% >|24.8% >|3.4% >|-34.6% >|18.4% >|16.3% >|14.5% >|14.9% >|24.4% >|24.3% >|31.9%
+|Average 11+^.^|*10.7%*
+|===
+
+== Conclusion
+
+For running OptaPlanner, the best garbage collector to use is still ParallelGC, which performance is even better in comparison with G1GC than it was in JDK11.
```

Comment:
Just a bit puzzled - if ParallelGC is so good why it works so bad with machine reassignment?
No doubt in numbers, I just not so familiar with machine reassignment specific.
Maybe it is the outcome of the fact that in this example "stop the world" strategy gets abusive so the feature turns into a bug?
Any recomendations on that?